Fix Portal Existing Hub vNet Requirement & Enable Wizard Mode

jtracey93 opened the pull request:
Fix Portal Existing Hub vNet Requirement & also enabled Wizard Mode (https://docs.microsoft.com/en-us/azure/azure-resource-manager/managed-applications/create-uidefinition-overview#wizard)

Test URL: https://portal.azure.com/#view/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fjtracey93%2Favdaccelerator%2Fjt-portal-test%2Fworkload%2Farm%2Fdeploy-baseline.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fjtracey93%2Favdaccelerator%2Fjt-portal-test%2Fworkload%2Fportal-ui%2Fportal-ui-baseline.json

cc: @danycontre for review 👍

danycontre commented:
THnaks for working on it Jack, my comments:

6: like isWizard mode

92-1015: nice solution to make hub vNet field required, there is an issue with the output being and object and not a string 

985: set selectAll to false (just to make it explicit)

1128: i see the module is now called avdExistingHubVnets, thinking about future feature to peer to multiple vNets? current output of this module is an object and breaks the templates beacuase they are waiting for a string, we can modify modules if it makes sense to account for more than 1 hub vnet input, will ask for additional feedback

cc: @nataliakon please review Jack's PR and comments to reach consensus and move foradward :)

jtracey93 commented:
Hey @danycontre,

Thanks for the review.

I have fixed the output for the hub VNET being an object, it is now a string. This was a typo on my side. 

- `avdExistingHubVnets` = the object of all subscriptions from the ARM API Control element
- `avdExistingHubVnet` = the selected subscription from the above list

I was using `avdExistingHubVnets` in the outputs where it should have been `avdExistingHubVnet` - have now fixed in last commit

Have also fixed the line 985 for the `selectAll` property value, this is now `false`
